[IV 21-22] Objetivo 6

Paszser opened the pull request:
Aclaro que hay algunos aspectos mejorables del Objetivo 5 los cuales, al no haberlo superado, se han dejado para refinar de cara a la convocatoria extraordinaria (sobre todo de documentación), aunque otros sí que se siguieron las correcciones y sí que se mejoraron. Aquí se abarcan los contenidos del objetivo 6.

* [x]  ¿Se han configurado varios sistemas de integración continua? -> Sí, tanto CircleCI como Github Actions.
* [x]  ¿Se ha configurado correctamente el *Checks API* en los sistemas en los que
   sea necesario para que aparezca correctamente en GitHub (y se pueda comprobar
   desde los tests)? -> Sí, se ha realizado en CircleCI y se puede comprobar en la documentación del mismo.
* [x]  ¿Uno de los sistemas configurados permite comprobar cuales son las versiones
   del lenguaje con las que funciona correctamente nuestra aplicación? -> Sí, se ha realizado en GithubActions y se puede comprobar en la documentación del mismo.
* [x]  ¿Se escogen de forma adecuada las versiones del lenguaje que se testean,
   tanto en el sistema de CI como en el contenedor Docker?

closes #28 
closes #29 
closes #30

Paszser committed:
Comienzo del PR Objetivo 6

Paszser committed:
Crea .circleci/config.yml avanzando #29

Paszser committed:
Crea docs/CI_CircleCI.md avanzando #29

Crea la documentación en la que se comentan aspectos del setup e importantes de la herramienta

Paszser committed:
Crea docs/CI.md avanzando #28

Crea la documentación en la que se comenta y justifica la elección de las herramientas usadas en el Objetivo

Paszser committed:
Modifica Rakefila automatizando lanzar el docker

Añadimos una opción que nos permita lanzar la imagen del docker automatizándolo.

Paszser committed:
Modifica README.md avanzando #28

Incluye un enlace en el README.md a la documentación referente a los sistemas de integración continua.

Paszser committed:
Añade clave CI a iv.yaml avanzando #29

Paszser committed:
Añade información al CI_CircleCI.md sobre checks

Paszser committed:
Crea CI_GithubActions.md avanzando #30

Paszser committed:
Crea gha.yml avanzando #30

Crea el archivo de configuración de Github Actions para ejecutar los tests pertinentes.

Paszser committed:
Corrige fallo de sintaxis en gha.yml

Corrige fallo de tabulación en la línea 12.

Paszser committed:
Corrige fallo de sintaxis en gha.yml

Corrige fallo de tabulación en la línea 14.

Paszser committed:
Corrige fallo de sintaxis en gha.yml

Corrige fallo de tabulación en la línea 15.

Paszser committed:
Corrige fallo de sintaxis en gha.yml avanzando #31

Corrige fallo de tabulación en la línea 8.

Paszser committed:
Corrige fallo de sintaxis en gha.yml avanzando #31

Corrige fallo de tabulación en la línea 4.

Paszser committed:
Crea directorio de imágenes en doc

Añade imagen de error de versiones de ruby

Paszser committed:
Corrige los fallos de sintaxis en gha.yml closes #31

Paszser committed:
Corrige versión de ruby en gha.yml avanzando #32

Paszser committed:
Corrige versión de ruby en gha.yml closes #32

Paszser committed:
Añade documentación de Github Action avanzando #30

Paszser committed:
Añade documentación de Github Action avanzando #30

Corrige dos enlaces

Paszser committed:
Modifica CI.md documentando las herramientas seleccionadas

JJ commented:
⛹ Revisores → @danifm1321 @lentes4k @Antobio17

lentes4k commented on docs/CI.md:
```diff
@@ -0,0 +1,57 @@
+
+En adición, me decanté por el uso de Github Actions en específico por un uso reciente en otro proyecto personal lo que me dio confianza para usarlo en esta ocasión.
+
+Otras herramientas abrían sido buenas opciones, véase Travis, Jenkins o Bamboo, pero la afinidad con las herramientas finalmente seleccionadas daba un plus de confianza a la hora de tomar una decisión.
```
typo -> habrían

Paszser replied:
Shame on me 😆 Juro que ha sido el nuevo teclado mecánico que tengo, no me acostumbro (?) 🤣

lentes4k commented on .circleci/config.yml:
```diff
@@ -0,0 +1,17 @@
+
+workflows:
+  version: 2
+  test-app-workflow:
```
creo que al tener la versión indicada arriba se puede prescindir en este punto. Pero no estoy seguro

Paszser replied:
He probado esa opción y me ha dado fallo, pero al eliminar el primero dejando la versión del workflows si que todo ha ido sobre ruedas 👌🏼

lentes4k commented on docs/CI_GithubActions.md:
```diff
@@ -0,0 +1,33 @@
+
+## Versiones de Ruby
+
+Aquí comentaremos un aspecto curioso que se ha dado en el desarrollo del archivo de configuración de Github Actions. A pesar de que la última versión estable de Ruby es la *3.1.0* como se puede ver [aquí](https://www.ruby-lang.org/es/downloads/) en el apartado de versiones estables, y siendo la 2.6.9 la última estable, siendo que más atrás ninguna tiene soporte. En la documentación seguida para e ...
```
más atrás me suena raro. Podrías poner algo así como "versiones anteriores" o del palo

lentes4k reviewed (approved):
Muy bien trabajado. Solo he detectado alguna minucia que te aparecerá por aquí abajito.
Como opinión personal creo que hay demasiada documentación en ciertos puntos. Está bien documentar aquello que hay propiciado conflicto o es reseñable, algunas otras cosas se pueden dar por supuestas.
Lo dicho, es mi opinión (realmente soy bastante malo documentando 🤣) y quizás sea una opinión equivocada. 

Dicho esto, por mi puedes seguir con los siguientes, ¡ánimo! 💯

Paszser committed:
Hace cambios solucionando #33 closes #33.

Se elimina la línea del archivo .rspec que daba error para ejecutar los tests.

Paszser committed:
Corrige fallo tipográfico en CI.md

Paszser commented:
> Muy bien trabajado. Solo he detectado alguna minucia que te aparecerá por aquí abajito. Como opinión personal creo que hay demasiada documentación en ciertos puntos. Está bien documentar aquello que hay propiciado conflicto o es reseñable, algunas otras cosas se pueden dar por supuestas. Lo dicho, es mi opinión (realmente soy bastante malo documentando 🤣) y quizás sea una opinión equivocada.
> 
> Dicho esto, por mi puedes seguir con los siguientes, ¡ánimo! 💯

¡Muchas gracias! Nada más vuelva a casa me pongo a revisar lo que me has señalado y refinarlo, porque tienes razón. Igualmente también voy aprendiendo poco a poco a documentar, espero poder hacerlo mejor paso a paso 😄 Cualquier duda que tenga luego te la pondré por acá.

Y nuevamente muchas gracias ^^.

Paszser committed:
Refina correciones afinando el objetivo.

Corrige clave version en el config.yml y en Github_Actions.md

Paszser committed:
Cambia clave version en config.yml

lentes4k commented:
> > Muy bien trabajado. Solo he detectado alguna minucia que te aparecerá por aquí abajito. Como opinión personal creo que hay demasiada documentación en ciertos puntos. Está bien documentar aquello que hay propiciado conflicto o es reseñable, algunas otras cosas se pueden dar por supuestas. Lo dicho, es mi opinión (realmente soy bastante malo documentando 🤣) y quizás sea una opinión equivocada.
> > Dicho esto, por mi puedes seguir con los siguientes, ¡ánimo! 💯
> 
> ¡Muchas gracias! Nada más vuelva a casa me pongo a revisar lo que me has señalado y refinarlo, porque tienes razón. Igualmente también voy aprendiendo poco a poco a documentar, espero poder hacerlo mejor paso a paso 😄 Cualquier duda que tenga luego te la pondré por acá.
> 
> Si por algún casual me pudieras dar algún ejemplo de documentación excesiva que veas o que dieras por supuesta te lo agradecería para poder entenderlo mejor y a partir de ahí quizás afinar el resto 😢 Como te dije, me cuesta barbaridad acertar al 100% con esto 😂 😂
> 
> Y nuevamente muchas gracias ^^.

Creo que donde quizás algo te hayas excedido es a la hora de documentar por separado los sitemas de integración continua. Pero ya te digo, yo no soy muy bueno documentando 😅

Paszser committed:
Refina CI.md dando últimos retoques

Paszser committed:
Modifica dockerhub.yml refinando aspectos del mismo.

Paszser committed:
Tratamos de resolver #34

Siguiendo los pasos en la documentación de CircleCI, tratamos de solucionar el issue.

Paszser committed:
Tratamos de resolver #34

Siguiendo los pasos en la documentación de CircleCI, tratamos de solucionar el issue.

Paszser committed:
Tratamos de resolver #34

Siguiendo los pasos en la documentación de CircleCI, tratamos de solucionar el issue.

Paszser committed:
Tratamos de resolver #34

Siguiendo los pasos en la documentación de CircleCI, tratamos de solucionar el issue.

Antobio17 reviewed (commented):
Mi revisión va un más general, ya que realmente la configuración inicial del sistema de integración continua no es "muy complicado" como has podido ver. Sin embargo creo que la documentación está demasiado enfocada a la teoría. Realmente con los requisitos que tenemos casi cualquier CI nos da buenos recursos para la integración en el proyecto. Por eso hubiera sido buena opción coger algunos sistemas, probar a configurar documentando por ejemplo con imágenes o trozos de código caracteristicos de la configuración y explicar facilidades/inconvenientes que te hubieras podido ir encontrando.

Faltaría también justificar las versiones de Ruby testeadas que es uno de los requisitos del objetivo 6.

Espero que te sirva mi punto de vista :+1:  cualquier cosa comenta!

Paszser committed:
Tratamos de resolver #34

Siguiendo los pasos en la documentación de CircleCI, tratamos de solucionar el issue.

Paszser committed:
Tratamos de resolver #34

Siguiendo los pasos en la documentación de CircleCI, tratamos de solucionar el issue.

Paszser committed:
Tratamos de resolver #34

Siguiendo los pasos en la documentación de CircleCI, tratamos de solucionar el issue.

Paszser committed:
Resolvemos fallo en config.yml closes #34

Paszser committed:
Refina apartado de justificación de versiones de Ruby

Antobio17 commented:
> Mi revisión va un más general, ya que realmente la configuración inicial del sistema de integración continua no es "muy complicado" como has podido ver. Sin embargo creo que la documentación está demasiado enfocada a la teoría. Realmente con los requisitos que tenemos casi cualquier CI nos da buenos recursos para la integración en el proyecto. Por eso hubiera sido buena opción coger algunos sistemas, probar a configurar documentando por ejemplo con imágenes o trozos de código caracteristicos de la configuración y explicar facilidades/inconvenientes que te hubieras podido ir encontrando.
> 
> Faltaría también justificar las versiones de Ruby testeadas que es uno de los requisitos del objetivo 6.
> 
> Espero que te sirva mi punto de vista +1 cualquier cosa comenta!

Fallo mío, no vi la justificación de las versiones en un issue y me lo salté sin querer en la documentación agregada.

Paszser commented:
> > Mi revisión va un más general, ya que realmente la configuración inicial del sistema de integración continua no es "muy complicado" como has podido ver. Sin embargo creo que la documentación está demasiado enfocada a la teoría. Realmente con los requisitos que tenemos casi cualquier CI nos da buenos recursos para la integración en el proyecto. Por eso hubiera sido buena opción coger algunos sistemas, probar a configurar documentando por ejemplo con imágenes o trozos de código caracteristicos de la configuración y explicar facilidades/inconvenientes que te hubieras podido ir encontrando.
> > Faltaría también justificar las versiones de Ruby testeadas que es uno de los requisitos del objetivo 6.
> > Espero que te sirva mi punto de vista +1 cualquier cosa comenta!
> 
> Fallo mío, no vi la justificación de las versiones en un issue y me lo salté sin querer en la documentación agregada.

¡No te preocupes! La verdad que intentando dejar todo más o menos completo no sé si al final está todo bien, me he pasado o no sé. Creo que al final en la documentación doy algunas razones para desechar algunas herramientas y otras razones para coger las finales, al menos voy haciendo lo que puedo a pesar del catarro que he pillado que apenas puedo respirar 😂 😂 

¡Muchas gracias por todo!

Paszser commented:
Creo que finalmente está preparado para revisión @JJ . Le recuerdo que puede haber algunos aspectos del Objetivo 5 que falte por repasar, pues dicho objetivo no lo pasé y en este me centré en los contenidos del 6. En la convocatoria extraordinaria me centraré en repasar el 5 y dejarlo listo. Muchas gracias por todo 😄

Paszser committed:
Puntualiza detalle sobre version en CircleCI

JJ commented on docs/CI.md:
```diff
@@ -0,0 +1,55 @@
+
+A continuación se va a presentar una lista de las opciones que se han barajado de cara a las dos herramientas a usar:
+
+* Jenkins: Servidor de automatización open source. Colabora en la automatización de parte del desarrollo software mediante la integración continua y la facilita. No es solo un motor con resultados binarios tales como "Success/Failed" si no que deja un feedback que mediante la webapp de la herramienta se puede visualizar. A su vez, Jenkins es una arquitectura de plugins, lo que la hace fácilemnte  ...
```
Si no vas a evaluar los 6 aspectos que has puesto antes, no los pongas.

JJ commented on docs/CI.md:
```diff
@@ -0,0 +1,55 @@
+
+### Justificación de la Elección
+
+Las principales razones para justificar la elección son que ambas herramientas cumplen con creces los requisitos impuestos con anterioridad. Ambas son compatibles con el testeo de varias versiones de nuestro lenguaje, el cual en este proyecto realizamos con Github Actions. Ambas son de uso libre y gratis, siendo así el nivel más básico de CircleCI, más que suficiente para este proyecto. Por ejemp ...
```
¿El resto no las cumplen?

JJ commented on docs/CI_GithubActions.md:
```diff
@@ -0,0 +1,35 @@
+
+Aquí comentaremos un aspecto curioso que se ha dado en el desarrollo del archivo de configuración de Github Actions. A pesar de que la última versión estable de Ruby es la *3.1.0* como se puede ver [aquí](https://www.ruby-lang.org/es/downloads/) en el apartado de versiones estables, y siendo la 2.6.9 la más antigua estable, y las versiones anteriores ninguna tiene soporte. En la documentación seg ...
+
+Por ello hemos optado por usar las versiones 3.0.0, y 3.0.1, pues son distintas a la misma que probamos en la imagen del docker y entran en los parámetros recomendados en la documentación de la herramienta.
```
Se prueban versiones para ver con cuantas es compatible. En el caso de Ruby sería imprescindible probar con la versión LTS de la 2. Precisamente se usa el CI para saber si es compatible o no.

JJ commented on .circleci/config.yml:
```diff
@@ -0,0 +1,16 @@
+      - checkout
+      - run:
+          name: "Task para app test"
+          command: "rake docker"
```
¿Dónde se configura Ruby? Si viene de serie, ¿por qué no se pone como justificación para elegir este y no otro?

JJ commented on .github/workflows/dockerhub.yml:
```diff
@@ -5,10 +5,6 @@ on:
     paths:
       - Dockerfile
       - Gemfile
```
¿Y si vaŕia Gemfile.lock?

JJ reviewed (changes requested):
* No se testean cosas diferentes con el contenedor docker y GH
* Se eligen los mismos que ha elegido (ya a estas alturas, y el 100% de los llegados después del periodo lectivo) el 99% de la clase, y sin justificar suficiente ni usar realmente los criterios escritos para elegirlos.
* No se justifican las versiones que se usan para testear.
